forem/forem
# Seed profile fields

## What type of PR is this? (check all applicable)


## Description

This PR explores how to add profile fields to new Forems. There are two constraints:

1. We want to seed different profile fields so admins don’t have to create everything from scratch. 
2. We don't want to add too many fields by default that admins have to remove/deactivate.

The possible solution explored in this PR creates different service classes for groups of fields. These get used in `db/seeds.rb` (where we probably just add all fields for local development), but also in other contexts like e.g. an internal admin controller.

## Related Tickets & Documents

Part of #6994

## QA Instructions, Screenshots, Recordings

1. Ensure you currently have no `ProfileField`s (in the DB): `ProfileField.destroy_all`.
2. Run `rails db:seed`.
3. You know should have 29 `ProfileField` records.

Alternatively, you can also call single classes directly, e.g. `ProfileFields::AddBaseFields.call`.

## Added tests?


## Added to documentation?


## [optional] Are there any post deployment tasks we need to perform?

## [optional] What gif best describes this PR or how it makes you feel?

![alt_text](gif_link)
